notebooks/delta_table_upload_demo.ipynb

```diff
@@ -0,0 +1,492 @@
+        "        config(\"spark.executor.extraClassPath\", f\"{SPARK_HOME}/jars/hadoop-aws-2.7.4.jar:{SPARK_HOME}/jars/aws-java-sdk-1.7.4.jar\")\n",
+        "\n",
+        "spark = configure_spark_with_delta_pip(builder).getOrCreate()\n",
+        "spark._jsc.hadoopConfiguration().set(\"fs.s3a.access.key\", AWS_ACCESS_KEY)\n",
+        "spark._jsc.hadoopConfiguration().set(\"fs.s3a.secret.key\", AWS_SECRET_KEY)\n",
+        "\n",
+        "results = client.create_data_rows_from_delta_table(dataset_id=dataset.uid, project_id=project.uid, verbose=True, table_path=table_path, spark=spark)"
+      ],
+      "metadata": {
+        "id": "RI13jipJ5-vO"
+      },
+      "execution_count": 13,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "Run `create_data_rows_from_delta_table()` with spark_config arg:"
+      ],
+      "metadata": {
+        "id": "5AH3hlr96zLi"
+      }
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "results = client.create_data_rows_from_delta_table(dataset_id=dataset.uid, project_id=project.uid, verbose=True, table_path=table_path, spark_config=aws_spark_config)"
+      ],
+      "metadata": {
+        "id": "Vpsssuy303g9"
+      },
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "#Check the status of the upload results to make sure the upload was successful\n",
+        "results"
+      ],
+      "metadata": {
+        "id": "50Wml9n1oKX5"
+      },
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "# GCS:\n",
+        "\n",
+        "\n",
+        "Build Spark Session before running `create_data_rows_from_delta_table()`:"
+      ],
+      "metadata": {
+        "id": "aDSYKdZ91xUR"
+      }
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "builder = SparkSession.\\\n",
+        "        builder.\\\n",
+        "        appName(\"pyspark-demo\").\\\n",
+        "        config(\"spark.driver.extraClassPath\", f\"{SPARK_HOME}/jars/gcs-connector-hadoop3-latest.jar\").\\\n",
+        "        config(\"spark.executor.extraClassPath\", f\"{SPARK_HOME}/jars/gcs-connector-hadoop3-latest.jar\")\n",
+        "\n",
+        "spark = configure_spark_with_delta_pip(builder).getOrCreate()\n",
+        "spark.conf.set(\"spark.hadoop.fs.gs.auth.service.account.enable\", \"true\")\n",
+        "spark.conf.set(\"google.cloud.auth.type\", \"SERVICE_ACCOUNT_JSON_KEYFILE\")\n",
+        "spark.conf.set('google.cloud.auth.service.account.json.keyfile', credentials_file)\n",
+        "spark.conf.set(\"fs.AbstractFileSystem.gs.impl\", \"com.google.cloud.hadoop.fs.gcs.GoogleHadoopFileSystem\")\n",
+        "\n",
+        "results = client.create_data_rows_from_delta_table(dataset_id=dataset.uid, verbose=True, table_path=table_path, spark=spark)"
```

Comment:
I was not able to get any batch_to_project_results to pass successfully. 
`{'data_row_upload_results': 'Success',
 'batch_to_project_results': ValueError("{'status': 'FAILURE', 'results': ['', '', '', ''], 'errors': [{'global_key': 'data_row', 'error': 'Data Row not found'}, {'global_key': 'project_id', 'error': 'Data Row not found'}, {'global_key': 'annotations', 'error': 'Data Row not found'}, {'global_key': 'dataset_id', 'error': 'Data Row not found'}]}"),
 'annotation_upload_results': []}`

Reply:
I am assuming a batch needs to be created in the project but this is not done anywhere in the code.